client/src/pages/ViewRequestStatus/RequestStatusView.js

```diff
@@ -11,6 +13,28 @@ export function RequestStatus() {
 
 	// this helps get the id from the router
 	const { id } = useParams();
+	useEffect(() => {
+		socket.on("connect", () => {
+			console.log("conncted");
+		});
+		socket.on("laptop_request:statusChanged", ({ laptopRequestId }) => {
+			setNeedsReloading((previousNeedsReloading) => {
+				return (
+					!previousNeedsReloading,
+					Notification.requestPermission().then(() => {
+						new Notification("Good news, we've found you a laptop!");
+					})
+				);
+			});
+
+			console.log("statusChanged", laptopRequestId);
+		});
+		socket.emit("test", { requestId: id });
+		return () => {
+			socket.off("connect");
+			socket.off("laptop_request:statusChanged");
+		};
+	}, [id, request]);
```

Comment:
don't include request in the use params -> otherwise it will be called twice!

Reply:
@gregdyke  I have deleted the request